printer/printast_with_mappings.ml

```diff
@@ -825,6 +825,7 @@ and module_expr i ppf x =
   attributes i ppf x.pmod_attributes;
   let i = i+1 in
   match x.pmod_desc with
+  | Pmod_apply_unit _ -> assert false (* XXX *)
```

Comment:
Putting a comment so we don't forget to fix it elsewhere before this is merged (there are probably not enough tests to trigger this assertion at runtime.